src/GameLogic/PlayerActions/Skills/SummonPartySkillPlugin.cs

```diff
@@ -0,0 +1,147 @@
+
+        try
+        {
+            for (var count = CountdownSeconds; count > 0; count--)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                foreach (var targetPlayer in targetPlayers)
+                {
+                    await targetPlayer.InvokeViewPlugInAsync<IChatViewPlugIn>(
+                        p => p.ChatMessageAsync($"Summoning in {count} second(s)...", player.Name, ChatMessageType.Party)).ConfigureAwait(false);
+                }
+
+                if (!player.IsAlive || player.IsAtSafezone())
+                {
+                    await player.Party.SendChatMessageAsync("Summoning canceled.", player.Name).ConfigureAwait(false);
+                    return;
+                }
+
+                await Task.Delay(1000, cancellationToken).ConfigureAwait(false);
+
+                targetPlayers = targetPlayers.Where(this.CanTargetBeTeleported);
+            }
+
+            await this.SummonTargetsAsync(player, targetPlayers).ConfigureAwait(false);
+        }
+        catch (OperationCanceledException)
+        {
+            // Handle cancellation (if needed)
+            await player.Party.SendChatMessageAsync("Summoning canceled.", player.Name).ConfigureAwait(false);
+        }
+        catch (Exception ex)
+        {
+            // Log unexpected exceptions
+            player.Logger.LogWarning(ex, "Error during countdown");
+        }
+        finally
+        {
+            player.SkillCancelTokenSource?.Dispose();
+            player.SkillCancelTokenSource = null;
+        }
+    }
+
+    private async ValueTask SummonTargetsAsync(Player player, IEnumerable<Player> targetPlayers)
+    {
+        foreach (var targetPlayer in targetPlayers)
+        {
+            Point targetPoint = player.Position;
+            bool foundValidPoint = false;
+            int maxAttempts = 10;
+            int attempts = 0;
+
+            while (!foundValidPoint && attempts < maxAttempts)
+            {
+                attempts++;
+
+                var offsetX = Rand.NextInt(-2, 3);
+                var offsetY = Rand.NextInt(-2, 3);
+                Point validPoint = new((byte)(player.Position.X + offsetX), (byte)(player.Position.Y + offsetY));
+
+                if (player.CurrentMap!.Terrain.WalkMap[targetPoint.X, targetPoint.Y]
+                    && player.Position.EuclideanDistanceTo(targetPoint) < 6)
+                {
+                    targetPoint = validPoint;
+                    foundValidPoint = true;
+                }
+            }
+
+            if (player.CurrentMap!.Definition.TryGetRequirementError(targetPlayer, out var errorMessage))
+            {
+                await targetPlayer.InvokeViewPlugInAsync<IShowMessagePlugIn>(p => p.ShowMessageAsync(errorMessage, Interfaces.MessageType.BlueNormal)).ConfigureAwait(false);
+                continue;
+            }
+
+            _ = Task.Run(() => targetPlayer.TeleportToMapAsync(player.CurrentMap!, targetPoint));
```

Comment:
No need for Task.Run here. But a try catch around that would be great, so that an error from one player is not affecting the next.

Reply:
Remvoed Task.Run. I did not add try catch since the TeleportToMapAsync already catches possible exceptions.